pkg/skaffold/output/output.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2021 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package output
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+)
+
+type SkaffoldWriter struct {
+	MainWriter  io.Writer
+	EventWriter io.Writer
+}
+
+func (s SkaffoldWriter) Write(p []byte) (int, error) {
+	n, err := s.MainWriter.Write(p)
+	if err != nil {
+		return n, err
+	}
+	if n != len(p) {
+		return n, io.ErrShortWrite
+	}
+
+	s.EventWriter.Write(p)
+
+	return len(p), nil
+}
+
+func SetupOutput(out io.Writer, defaultColor int, forceColors bool) io.Writer {
```

Comment:
nit: How about `output.Setup` instead of `output.SetupOutput` to reduce stutter?

super nit: `Setup` seems to suggest the method has side-effects, but if it's only returning the type could it be called `GetWriter` or something?